File: years/2020/day06/part1.py
def main():

    with open("input.txt") as handle:
        input = handle.read().splitlines()

    count = 0

    lines = input

    answers = set()

    for line in lines:
        match line:
            case "": # end of a group
                count += len(answers)
                answers = set()
            case _:
                answers.update(iter(line))
    else:
        count += len(answers)
        answers = set()


    print(count)


main()

File: years/2024/day02/src/bin/part1.rs
use std::fs;

fn check_record(record: &str) -> bool {
    let mut previous: Option<i32> = None;
    let mut ascending: Option<bool> = None;

    for level in record.split_whitespace().map(|x| x.parse::<i32>().unwrap()) {
        if let Some(prev) = previous {
            let diff = level - prev;
            match ascending {
                None => {
                    ascending = Some(diff.is_positive());
                }
                Some(true) => {
                    if diff.is_negative() {
                        return false;
                    }
                }
                Some(false) => {
                    if diff.is_positive() {
                        return false;
                    }
                }
            }
            if diff.abs() < 1 || diff.abs() > 3 {
                return false;
            }
        }
        previous = Some(level);
    }

    true
}

fn main() {
    let input = fs::read_to_string("input.txt").unwrap();
    let lines: Vec<&str> = input.lines().collect();
    let cnt = lines.iter().filter(|x| check_record(x)).count();
    println!("{}", cnt);
}
